scripts/iam-tools/aws_creds_check.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+
+# Note: because we source this in the login shell, it needs to be compatible with bash and zsh.
+# There are several places where -n "$ZSH_NAME" is tested to see if we're running zsh, and then
+# the zsh version of the syntax is used instead of the bash version.
+
+# only bash v4+ and zsh are supported. do nothing for other shells
+if [ -n "$BASH_VERSION" -a "$BASH_VERSINFO" -gt 3 -o "$ZSH_NAME" ]; then
+
+function _check_creds()
+{
+    local -A accounts
+    local account_count creds_actual name num line regex now credfile=~/.aws/credentials expfile=~/.aws/credentials_expiration
+
+    if ! private_encfs_mounted; then
+        echo "~/.private not mounted. Skipping AWS credential check"
+        return
+    fi
+
+    # if ~/.private/.aws doesn't exist, create it
+    test -d ~/.private/.aws || mkdir ~/.private/.aws
+    # if ~/.private/.aws/credentials doesn't exist, create it
+    test -f ~/.private/.aws/credentials || { touch ~/.private/.aws/credentials; chmod 600 ~/.private/.aws/credentials; }
```

Comment:
For the sake of readability, I recommend using the conditional test syntax.  Also, avoid putting multiple statements on a single line.

`if [ ! -f /the/creds ]; then`
` touch /the/creds `
` chmod 600 /the/creds `
`fi`